File: Models/Category.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace dotnet_backend_freshmart.Models
{
    public class Category
    {
        public Guid Id { get; set; }
        public string Slug { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string? Icon { get; set; }
        public short SortOrder { get; set; } = 0;
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Quan hệ 1 Danh mục có nhiều Sản phẩm
        public virtual ICollection<Product> Products { get; set; } = new List<Product>();
    }
}

File: freshmart/src/types/category.ts
// ============================================================================
// Types cho module Danh mục hàng hóa (Categories)
// Khớp 100% với DTOs của Backend ASP.NET Core
// ============================================================================

export interface Category {
  id: string;
  slug: string;
  name: string;
  icon?: string | null;
  sortOrder: number;
  isActive: boolean;
  productCount: number;
  createdAt: string;
}

export interface CreateCategoryRequest {
  name: string;
  slug?: string;
  icon?: string;
  sortOrder?: number;
  isActive?: boolean;
}

export interface UpdateCategoryRequest {
  name: string;
  slug?: string;
  icon?: string;
  sortOrder?: number;
  isActive?: boolean;
}
